src/components/password-strength-field/password-strength.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, Input, SimpleChanges } from '@angular/core';
+import { PasswordService } from '../../services/password-service/password-service';
+import { BulkTranslateService } from '../../services/bulk-translate-service/bulk-translate-service';
+import { TranslateService } from '@ngx-translate/core';
+
+/**
+ * Generated class for the PasswordStrengthComponent component.
+ *
+ * See https://angular.io/api/core/Component for more info on Angular
+ * Components.
+ */
+@Component({
+  selector: 'smilo-password-strength-field',
+  templateUrl: 'password-strength.html'
+})
+export class PasswordStrengthComponent {
+  // The password input to check
+  @Input() password: string;
+  // List of translations set programmatically
+  translations: Map<string, string> = new Map<string, string>();
+  // The password strength object
+  passwordStrength: any;
```

Comment:
Just to verify but the library used does not have any Typescript typings online? If not is the passwordStrength object very complicated? If not we might be able to add a simple interface definition ourselves.